README.md

```diff
@@ -8,7 +8,152 @@ This repository serves as a quickstart template for GDExtension development with
 * GitHub CI/CD to publish your library packages when creating a release (`.github/workflows/builds.yml`)
 * preconfigured source files for C++ development of the GDExtension (`src/`)
 
-## Usage
+## Usage - Actions
+
+The actions builds `godot-cpp` at a specified location, and then builds the `gdextension` at a configurable location. It builds for desktop, mobile and web and allows for configuration on what platforms you need. It also supports configuration for debug and release builds, and for double builds.
+
+The action uses SConstruct for both godot-cpp and gdextension that is built.
+
+To reuse the build actions, in a github actions yml file, do the following:
+
+```name: Build GDExtension
+on:
+  workflow_call:
+  push:
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - platform: linux
+            arch: x86_64
+            os: ubuntu-20.04
+          - platform: windows
+            arch: x86_32
+            os: windows-latest
+          - platform: windows
+            arch: x86_64
+            os: windows-latest
+          - platform: macos
+            arch: universal
+            os: macos-latest
+          - platform: android
+            arch: arm64
+            os: ubuntu-20.04
+          - platform: android
+            arch: arm32
+            os: ubuntu-20.04
+          - platform: android
+            arch: x86_64
+            os: ubuntu-20.04
+          - platform: android
+            arch: x86_32
+            os: ubuntu-20.04
+          - platform: ios
+            arch: arm64
+            os: macos-latest
+          - platform: web
+            arch: wasm32
+            os: ubuntu-20.04
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: 🔗 GDExtension Build
+        uses: ughuuu/godot-cpp-template/.github/actions/build@main
+        with:
+          platform: ${{ matrix.platform }}
+          arch: ${{ matrix.arch }}
+          godot-cpp-location: godot-cpp
+          float-precision: single
+          build-target-type: template_release
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: GDExtension
+          path: |
+            ${{ github.workspace }}/bin/**
+
+```
+
+## Usage - Signing
+
+You will need:
+
+- A macbook
+- An apple id enrolled in apple developer(99 dollar per year)
+- In the framework folder, you need to create a `Resources/Info.plist`. Take as example the one in this project. Be careful to set CFBundleExecutable to the EXACT lib name, otherwise it won't work. Also, don't put strange names in the CFBundleName and other such places, try to only use letters and space. Errors will be if not extremly vague to impossible to debug.
+
+For the actions you will need to set the following inputs. Store them as secrets in github:
```

Comment:
```suggestion
- A Mac
- An Apple ID enrolled in Apple Developer Program (99 USD per year)
- A `Resources/Info.plist` in the `framework` folder. Take the one in this project as an example. Be careful to set CFBundleExecutable to the **EXACT** lib name, otherwise it won't work. Also, don't put strange names in the CFBundleName and other such places. Try to only use letters and spaces. Errors will be extremly vague if not impossible to debug.

For the actions you will need to set the following inputs. Store them as secrets in GitHub:
```